commonmark/src/main/java/org/commonmark/parser/InlineParser.java

```diff
@@ -14,4 +13,9 @@
      */
     void parse(String input, Node node);
 
+    /**
+     * @return how many characters were parsed as a reference, {@code 0} if none
+     */
+    int parseReference(String s);
```

Comment:
We've looked at the intended behavior behind parseReference and agree that it shouldn't be part of the InlineParser's API.
But it looks like the ParagraphParser uses this method when calling `closeBlock`.

We've searched through the code and we're not sure how to best divorce `parseReference` from the InlineParser interface when it's relied upon by ParagraphParser, given that the ParagraphParser is required to update the internal state/referenceMap of the InlineParser.

Any ideas here?